docs/decisions/blue-green-deployments.md

```diff
@@ -0,0 +1,190 @@
+
+Sometimes it is necessary to have multiple versions of a service running in parallel so that the new version can be initialized and / or tested while the old version is still available. This is commonly referred to as [Blue-green deployment](https://en.wikipedia.org/wiki/Blue%E2%80%93green_deployment).
+
+The typical steps involved in a deployment would be as follows:
+
+1. Deploy new green version (old blue version is still running)
+2. Perform any tests and initializations on the green version
+3. Update other services to reference the green version where they were previously referencing the blue version
+4. Take down the blue version
+
+In the subsequent release blue and green would be swapped above.
+
+This decision record considers how blue-green deployments should be managed.
+
+## Decision Drivers
+
+* it should be easy to determine which is the current deployment (blue or green)
+* it must be possible to test the deployment process locally
+* it may be desirable to avoid parallel deployments in non-production environments / locally when resources are constrained and a period of down time is acceptable
+* it should be easy to switch other services to reference the new version once testing / initialization has been completed
+
+## Considered Options
+
+There are two aspects to this decision:
+
+* How to deploy the alternative service versions
+* Switching between the two deployments
+
+### Deployment Mechanism
+
+* Deploy different versions on separate hosts
+* Deploy the two versions as separate docker compose projects
+* Manage the different versions within a single docker compose project
+
+### Switching Between Deployments
+
+* Use a shared reverse proxy
+* Use a reverse proxy on each project
+* Target version is configured independently for each referencing service
+* Re-deploy the target service with the normal ports re-mapped
+
+## Decision Outcome
+
+Chosen option is to "Manage the different versions within a single docker compose project" and "Re-deploy the target service with the normal ports re-mapped" as this will work equally well for local and production deployments and does not require other services to be re-configured during the transition.
+
+The short period of downtime during the switch from one version to the next would be no different than for normal deployments.
+
+### Consequences
+
+The chosen option is not strictly a "blue-green" approach but more of a "current-next" option with versioned storage. An example of the steps needed to make a transition is provided below:
+
+An existing `example` service with one volume that maps its internal port 80 to 8080 on the host, e.g.:
+
+```
+services:
+  example:
+    image: ghcr.io/openfoodfacts/example:${TAG}
+    build: .
+    ports:
+      - 8080:80
+    volumes:
+      - dbdata:/var/data
+```
+
+The first step in deploying a new version would be to add a new service and volume and update the old service to use the last build image, e.g.
```

Comment:
You don't explicitly tell how you intend those changes to be done.

Personally, for this kind of migration, it seems ok to me to do them directly editing files in production. This avoids a lot of latency and even eventual pitfalls due to CI.